### PR TITLE
Update AEM Forms solutions with new names.

### PR DESCRIPTION
**Pull request by arnehjain**
# Update AEM Forms include flags to reflect new solution types

## Description
Going forward, AEM Forms will have two different solutions on
AEM as a Cloud Service. These are named:
 - formsenrollment
 - formscommunications

As per the existing contract with Cloud Manager during VSTS repository
generation, the updated solution names will be passed instead of the
earler used `includeForms` in the archetype command.

This requires us to update all usage of the `includeForms` property within
the archetype with the updated solution names (`includeFormsenrollment`
and `includeFormscommunications`).

For backward compatibility reasons the older `includeForms` check is
left intact in the archetype template but has been removed from the
README.md.

## Related Issue
#823

## Motivation and Context

This is needed to preserve the contract with Cloud Manager and archetype for some new AEM Forms Features.
## How Has This Been Tested?
Run build-time tests and validate output.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by GuillaumeCleme** [comment]
Can we also update the ReadMe description for the `includeForms` argument? It currently describes the argument as the following but this description does not make any mention of AEM Forms which is a product module that is very different than Core Components. 

> Includes Forms Core Components dependencies and generates corresponding artifacts.

Arguably, AEM Forms and the `includeForms` should be different arguments altogether. I would suggest `includeCoreForms`, `includeFormsEnrollment`, and `includeFormsCommunications` if the last two have been vetted by the Product Team.

**Comment by arnehjain** [comment]
@GuillaumeCleme, Apologies for the late reply. It was an extended holiday over here.

Let me try to provide a little more context. 

So far till now, there was a single "AEM Forms" solution in AEM as a Cloud Service and it was just named "Forms". However going forward from this month onwards, this solution is now named "Forms Enrollment" and there will be an additional solution named "Forms Communications". 
So in total, there will be 2 Forms specific solutions on AEM as a Cloud Service parallel to other solutions like Sites, Assets, etc. 
- Forms Enrollment. ( -DincludeFormsenrollment)
- Forms Communications ( -DincludeFormscommunications)

However, for the short term, for backward compatibility reasons we'd still need to support the older "Forms" solution which has the flag ( -DincludeForms). Once these older solutions have been migrated to the newer ones, we should be able to remove `includeForms` from the archetype.

> Arguably, AEM Forms and the includeForms should be different arguments altogether. I would suggest includeCoreForms, includeFormsEnrollment, and includeFormsCommunications if the last two have been vetted by the Product Team

`includeFormsenrollment and `includeFormscommunications` have been vetted by the Product Team (Both CMGR and AEM Forms). `includeForms` is not needed anymore and is only kept for backward compatibility reasons. Unfortunately we're not in a position to change the naming conventions wrt the name casing at this point (since the contract between CMGR and the archetype is pre-defined)

File: src/main/resources/META-INF/archetype-post-generate.groovy [hunk 1]
```diff
@@ -25,6 +25,8 @@ def sdkVersion = request.getProperties().get("sdkVersion")
 def includeDispatcherConfig = request.getProperties().get("includeDispatcherConfig")
 def includeCommerce = request.getProperties().get("includeCommerce")
 def includeForms = request.getProperties().get("includeForms")
```
Comment (rismehta) [inline src/main/resources/META-INF/archetype-post-generate.groovy:27]: Why can't we just use the same definition includeForms here ?
```
def includeForms = "n"
        request.getProperties().entries().each { item ->
            if (item.getKey().equalsIgnoreCase("includeforms") && item.getValue().equalsIgnoreCase("y")) {
                includeForms = "y"
            } ....
        }
```

Comment (arnehjain) [reply src/main/resources/META-INF/archetype-post-generate.groovy:27]: That is possible but we'd need to change it again when `includeForms` is removed in the near future. The current version is consistent with changes elsewhere as well. I can make this change if required though.

**Review by rismehta — APPROVED** [review]

**Pull request merged**